ui/src/time_picker.test.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright 2020 H2O.ai, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { fireEvent, render, act, waitFor } from '@testing-library/react'
+import React from 'react'
+import { TimePicker, XTimePicker } from './time_picker'
+import { wave } from './ui'
+
+const
+  name = 'timepicker',
+  timepickerProps: TimePicker = { name, label: 'Select time' }
+
+const waitForIdleEventLoop = async () => act(() => new Promise(res => setTimeout(res)))
+
+describe('time_picker.tsx', () => {
+
+  beforeEach(() => {
+    wave.args[name] = null
+  })
+
+  it('Renders data-test attr - lazy load placeholder', async () => {
+    const { getByTestId } = render(<XTimePicker model={timepickerProps} />)
+    await waitForIdleEventLoop()
+    expect(getByTestId('lazyload')).toBeInTheDocument()
+  })
+
+  it('Renders data-test attr - time picker component', async () => {
+    const { getByTestId } = render(<XTimePicker model={timepickerProps} />)
+    await waitForIdleEventLoop()
+    await waitFor(() => expect(getByTestId(name)).toBeInTheDocument())
+  })
+
+  it('Sets args - init - value not specified', async () => {
+    render(<XTimePicker model={timepickerProps} />)
+    await waitForIdleEventLoop()
+    expect(wave.args[name]).toBeNull()
+  })
+
+  it('Sets args - init - value specified', async () => {
+    render(<XTimePicker model={{ ...timepickerProps, value: '10:30' }} />)
+    await waitForIdleEventLoop()
+    expect(wave.args[name]).toBe('10:30')
+  })
+
+  it('Show correct input value in 12 hour time format', async () => {
+    const { getByDisplayValue } = render(<XTimePicker model={{ ...timepickerProps, value: '14:30' }} />)
+    await waitForIdleEventLoop()
+    expect(getByDisplayValue('02:30 PM')).toBeInTheDocument()
+  })
+
+  it('Shows midnight correctly in 12 hour time format', async () => {
```

Comment:
I would also add a an expect for `wave.args` to all these kind of tests to check if `wave.args` are still submitted in 24h format despite visual format being 12.